### PR TITLE
Handle undefined __git_complete

### PR DESCRIPTION
**Pull request by darcyparker**
Addresses https://github.com/jonas/tig/issues/1011

As identified in https://github.com/jonas/tig/issues/1011#issuecomment-736922515, there are cases where `__git_complete` is undefined because it is loaded dynamically by bash_completions.

As suggested in https://github.com/jonas/tig/issues/1011#issuecomment-736938541, using `pkg-config` to find `git-completion` script, which defines `__git_complete`.  And using approach similar to https://github.com/felipec/git-completion/blob/master/git-completion.zsh#L28-L49

### CODE REVIEW

File: contrib/tig-completion.bash [hunk 1]
```diff
@@ -29,6 +29,27 @@
 #       at source time then all lookups will be done on demand,
 #       which may be slightly slower.
 
+#tig-completion requires __git_complete
+#* If not defined, source git completions script so __git_complete is available
+if ! declare -f __git_complete &>/dev/null; then
+  _bash_completion=$(pkg-config --variable=completionsdir bash-completion 2>/dev/null) ||
```
Comment (rohieb) [inline contrib/tig-completion.bash:35]: Nitpick: the indentation should be consistent with the rest of the file.

Comment (darcyparker) [reply contrib/tig-completion.bash:35]: Thank you. I will fix that.

**Comment by rohieb** [comment]
Yep, LGTM now.

**Comment by koutcher** [comment]
Thanks.

**Pull request merged**